commit 9a5e5fdbccddb16b815ded9c50f7de42bfaa1f5e
Author: rhb2 <robert.bogart@joyent.com>
Date:   2019-09-04T18:57:12+00:00 (5 weeks ago)
    
    MANTA-4531 Empty failed task list returned from agent appears as Some([]) instead of None

diff --git a/src/agent.rs b/src/agent.rs
index cead4db..135773c 100644
--- a/src/agent.rs
+++ b/src/agent.rs
@@ -741,8 +741,13 @@ fn process_assignment(assignments: Arc<Mutex<Assignments>>, uuid: String) {
         tmp.tasks[i] = t;
     }
 
+    let failed = match failures.len() {
+        0 => None,
+        _ => Some(failures),
+    };
+
     assignment.write().unwrap().stats.state =
-        AgentAssignmentState::Complete(Some(failures));
+        AgentAssignmentState::Complete(failed);
     assignment_complete(assignments, uuid);
 }
 
